spec/syntax/parser_spec.cr

```diff
@@ -527,7 +527,21 @@ describe "Parser" do
   # Assignments are not allowed to methods with modifiers
   it_does_not_parse %q(a.b? = 1)
   it_does_not_parse %q(a.b! = 1)
-
+  # Assigned Anonymous Functions called should be coerced to a Call
+  it_parses %q(
+    foo = fn 
+            ->() { } 
+          end
+    foo()
+  ), SimpleAssign.new(Var.new("foo"), AnonymousFunction.new([Block.new])), Call.new(nil, "foo")
```

Comment:
Just for consistency, I'd replace uses of `Var.new("name")` with `v("name")`.

There are a bunch of helpers defined in [`spec/support/nodes.cr`](https://github.com/myst-lang/myst/blob/master/spec/support/nodes.cr) that are shorthand for creating Node instances. I think it would help readability a bit here.